runtime/nativeMain/src/kotlinx/benchmark/native/NativeExecutor.kt

```diff
@@ -1,73 +1,172 @@
 package kotlinx.benchmark.native
 
 import kotlinx.benchmark.*
+import kotlin.native.internal.GC
 import kotlin.system.*
 
 class NativeExecutor(name: String, args: Array<out String>) : SuiteExecutor(name, args) {
+
+    data class BenchmarkRun(val benchmarkName: String, val config: BenchmarkConfiguration,
+                            val parameters: Map<String, String>)
+
+    private fun outputBenchmarks(runnerConfiguration: RunnerConfiguration,
+                                 benchmarks: List<BenchmarkDescriptor<Any?>>,
+                                 start: () -> Unit) {
+        start()
+        benchmarks.forEach {
+            val suite = it.suite
+            val config = BenchmarkConfiguration(runnerConfiguration, suite)
+            var parametersId = 0
+            runWithParameters(suite.parameters, runnerConfiguration.params, suite.defaultParameters) { params ->
+                val benchmarkRunConfig = buildString {
+                    appendLine("benchmark: ${it.name}")
+                    appendLine("configuration: $config")
+                    appendLine("parameters: $params")
+                }
+                val fileName = "${additionalArguments[1]}/${suite.name}_${it.name}_$parametersId.txt"
+                writeFile(fileName, benchmarkRunConfig)
+                parametersId++
+            }
+        }
+    }
+
+    private fun List<BenchmarkDescriptor<Any?>>.getBenchmark(name: String) = find { it.name == name }
+        ?: throw NoSuchElementException("Benchmark $name wasn't found.")
+
+    private fun runBenchmarkIteration(benchmarks: List<BenchmarkDescriptor<Any?>>) {
+        val (configFileName, iteration, resultsFile) = additionalArguments
+        val benchmarkRun = configFileName.parseBenchmarkConfig()
+        val benchmark = benchmarks.getBenchmark(benchmarkRun.benchmarkName)
+        val samples = run(benchmark, benchmarkRun, iteration.toInt())
```

Comment:
Does each measurment iteration run a warmup iteration first?

Reply:
Yes, because without extra single run we can't calculate iteration number in case if time is less then set iterationTime